java/openpkw-services/src/main/java/org/openpkw/services/CsvParser.java

```diff
@@ -0,0 +1,37 @@
+package org.openpkw.services;
+
+import org.springframework.batch.item.file.LineMapper;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Created by wojciech.szostak on 15.08.2015.
+ */
+public class CsvParser<T>
+{
+    private LineMapper lineMapper;
+
+    public CsvParser(LineMapper<T> lineMapper) {
+        this.lineMapper = lineMapper;
+    }
+
+    public List<T> parse(File file) throws Exception
+    {
+        List<T> result = new ArrayList<>();
+
+        BufferedReader br = new BufferedReader(new FileReader(file));
+
+        int lineNumber = 0;
+        for(String line; (line = br.readLine()) != null; ) {
```

Comment:
line = br.readLine()) != null;  
To powinno iść do nowej metody z odpowiednią nazwą. Poza tym myślę, że powinniśmy użyć tu jakiejś klasy do Stringów, chociażby StringUtils.isEmpty() z apacha. Sprawdza od razu czy String jest pusty i czy jest nullem.